src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx

```diff
@@ -31,40 +31,67 @@ const valueRestrictions = {
   p_val_adj: [0, 1],
 };
 
-const presetFilters = {
-  'Up-regulated': {
+const presetFilters = [
+  {
+    label: 'Up-regulated',
     columnName: 'logFC',
     comparison: 'greaterThan',
     value: 0,
-  },
-  'Down-regulated': {
+  }, {
+    label: 'Down-regulated',
     columnName: 'logFC',
     comparison: 'lessThan',
     value: 0,
-  },
-  Significant: {
+  }, {
+    label: 'Significant',
     columnName: 'p_val_adj',
     comparison: 'lessThan',
     value: 0.05,
   },
-};
+];
 
 const AdvancedFilteringModal = (props) => {
   const { onCancel, onLaunch } = props;
+
+  const [availablePresetFilters, setAvailablePresetFilters] = useState(presetFilters);
+  const [availableCriteriaOptions, setAvailableCriteriaOptions] = useState(criteriaOptions);
+
   const [form] = Form.useForm();
   const advancedFilters = useSelector((state) => (
     state.differentialExpression.comparison.advancedFilters)) || [];
-  const diffExprLoading = useSelector((state) => state.differentialExpression.properties.loading);
+  const {
+    loading: diffExprLoading,
+    data: diffExprData,
+  } = useSelector((state) => state.differentialExpression.properties);
+
+  useEffect(() => {
+    if (!diffExprData[0]) return;
+
+    const availableColumns = Object.keys(diffExprData[0]);
+
+    const filteredPresetFilters = presetFilters
+      .filter((filter) => availableColumns.includes(filter.columnName));
+
+    const filteredCriteriaOptions = criteriaOptions
+      .filter((option) => availableColumns.includes(option.value));
+
+    console.log('*** filteredPresetFilters', filteredPresetFilters);
+    console.log('*** filteredCriteriaOptions', filteredCriteriaOptions);
+
+    setAvailablePresetFilters(filteredPresetFilters);
+    setAvailableCriteriaOptions(filteredCriteriaOptions);
+  }, [diffExprData.length]);
```

Comment:
What if the diffexprdata changes but the length is the same, will this retrigger?

Reply:
Oh right, this won't retrigger. This should retrigger if the number of column changes. I'll update this.